### PR TITLE
Created Fast Jar test for Gradle

### PR DESCRIPTION
**Pull request by Garima829**
Gradle equivalent of testThatFastJarFormatWorks()

### CODE REVIEW

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        GradleRunner.create()
+                .forwardOutput()
+                .withPluginClasspath()
+                .withArguments(arguments("package", "DskipTests","-Dquarkus.package.type=fast-jar"))
+                .withProjectDir(projectDir)
+                .build();
+
+//        await().atMost(1, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
+//        assertThat(running.log()).containsIgnoringCase("BUILD SUCCESS");
+//        running.stop();
```
Comment (geoand) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:50]: Please remove this if it's not going to be used.

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        fail("Failed to assert that the application properly reads resources from the classpath");
+    }
+
+    // we don't use REST Assured because its bundled Groovy version clashes with Maven Invoker's (which is also used in this module)
+    static void assertApplicationPropertiesSetCorrectly() {
+        assertApplicationPropertiesSetCorrectly("");
+    }
+
+    // we don't use REST Assured because its bundled Groovy version clashes with Maven Invoker's (which is also used in this module)
+    static void assertApplicationPropertiesSetCorrectly(String prefix) {
```
Comment (geoand) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:121]: Same here

**Review by geoand — COMMENTED** [review]
Great first try!

I have added some comments

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java (hunk 1 above)
Comment (Garima829) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:50]: I wasn't able to replicate this part from the Maven test, do we actually need it here for testing the time limit?

Comment (geoand) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:50]: Nope, we don't need it because it's handled automatically by Gradle

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java (hunk 2 above)
Comment (geoand) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:121]: Darn, it seems like this comment is out of context:

I meant that I am pretty sure our Gradle tests have a much better way of doing this test. See method `getHttpResponse` from `QuarkusDevGradleTestBase` (which should probably be moved to `QuarkusGradleTestBase`)

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java (hunk 1 above)
Comment (Garima829) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:50]: Okay

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java (hunk 2 above)
Comment (Garima829) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:121]: On it

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        commands.add(JavaBinFinder.findBin());
+        commands.add("-jar");
+        commands.add(jar.toString());
+        ProcessBuilder processBuilder = new ProcessBuilder(commands.toArray(new String[0]));
+        processBuilder.redirectOutput(output);
+        processBuilder.redirectError(output);
+        return processBuilder.start();
+    }
+
+    static void assertResourceReadingFromClassPathWorksCorrectly(String path) {
```
Comment (geoand) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:90]: This should probably also be replaced with the proper `getHttpResponse`

Comment (Garima829) [reply devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java:90]: Okayy, will do

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/BasicJavaPlatformModuleTest.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
     @Test
     public void testBasicPlatformModuleBuild() throws Exception {
 
-        final File projectDir = getProjectDir("basic-java-platform-module");
+        final File projectDir = getProjectDir("basic-java-p    latform-module");
```
Comment (geoand) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/BasicJavaPlatformModuleTest.java]: This change doesn't look right...

Comment (Garima829) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/BasicJavaPlatformModuleTest.java]: Yes, editing it

File: integration-tests/gradle/src/test/resources/test-that-fast-jar-format-works/src/main/resources/META-INF/resources/index.html [hunk 5]
```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
```
Comment (geoand) [inline integration-tests/gradle/src/test/resources/test-that-fast-jar-format-works/src/main/resources/META-INF/resources/index.html:1]: This file can be deleted

File: pom.xml [hunk 6]
```diff
@@ -31,7 +31,7 @@
     </scm>
 
     <properties>
-        <graalvmHome>${env.GRAALVM_HOME}</graalvmHome>
+        <graalvmHome>$GRAALVM_HOME</graalvmHome>
```
Comment (geoand) [inline pom.xml:34]: Why were these changed?

Comment (Garima829) [reply pom.xml:34]: It was giving some error, I changed these before the first commit i guess, should I undo this?

File: integration-tests/gradle/src/test/resources/test-that-fast-jar-format-works/src/main/resources/META-INF/resources/index.html (hunk 5 above)
Comment (Garima829) [reply integration-tests/gradle/src/test/resources/test-that-fast-jar-format-works/src/main/resources/META-INF/resources/index.html:1]: On it

File: pom.xml (hunk 6 above)
Comment (geoand) [reply pom.xml:34]: Yes please

Comment (Garima829) [reply pom.xml:34]: Okk

**Review by geoand — APPROVED** [review]
Thanks! 

Please the PR out of draft so CI can run on it

**Comment by geoand** [comment]
Please squash the commits
